Mn tc welcome screen styling

Tanzi11 opened the pull request:
##  Description

We were tasked to create an aesthetically pleasing welcome screen. Inspired by Material-UI, we wanted an interface that was intuitive and engaging. Experimenting with color palettes and user navigation elements, we bring to you the welcome screen. 

We first implemented Material-UI but found customization a bit difficult to grasp and use in the amount of time to complete the issue and went with React Bootstrap as it's a dependency already installed.

In prototyping different color palettes, we wanted to keep accessibility in mind.  This [article](https://stephaniewalter.design/blog/color-accessibility-tools-resources-to-design-inclusive-products/) was a great resource for different accessibility tools to check color and contrast.

We used this [tool](https://whocanuse.com/?b=ffffff&c=007f4f&f=16&s=) to check color visibility and contrast for users different types of color blindness and limited vision, providing the contrast ratio and WCAG grading comparison of two different colors.  The contrast ratio is the difference in luminous or color that makes things like text or other objects distinguishable and this affects the accessibility grading of websites with
A being the minimum level of accessibility
AA being acceptable level (contrast ratio is 4.5) and
AAA being the gold standard (contrast ratio of 7).

Even chrome dev tools will show the contrast ratio of colors used in an element
<img width="374" alt="Screen Shot 2020-06-19 at 11 45 07 PM" src="https://user-images.githubusercontent.com/2355543/85198369-0911ea00-b29d-11ea-8b5b-d877020ae902.png">

[This site](https://toolness.github.io/accessible-color-matrix/?n=white&n=Color%202&n=Color%203&n=Color%204&n=Color%205&n=Color%206&v=FFFFFF&v=456102&v=C6E2B2&v=F9593A&v=FCCE38&v=000000) was a great also resource to check contrast ratios on a palette of colors, pointing out which colors should and shouldn't be used together to pass accessibility grades.

## Prototypes

![Screen Shot 2020-06-18 at 5 11 12 PM](https://user-images.githubusercontent.com/40173387/85180282-0ab5c080-b251-11ea-813a-7507e16c7b50.png)

![Screen Shot 2020-06-20 at 3 10 01 PM](https://user-images.githubusercontent.com/40173387/85209734-46f82800-b308-11ea-820a-c4589d9946a4.png)

## Related Issue

closes #15 

## Acceptance Criteria

- The welcome screen (the screen a user sees before they’ve created their list) looks something like the following (or equivalent, use your creativity!):

## Type of Changes

<!-- Put an `✓` for the applicable box: -->

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|  ✓ | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before

<!-- If UI feature, take provide screenshots -->
![prev-welcome-screen](https://user-images.githubusercontent.com/2355543/85197935-4b392c80-b299-11ea-8fec-5e572cdff9f8.png)

### After

<!-- If UI feature, take provide screenshots -->
![welcom-screen](https://user-images.githubusercontent.com/2355543/85197899-fc8b9280-b298-11ea-94d5-5b7c607fdefa.png)


## Testing Steps / QA Criteria

<!-- Provide steps the other cohort members and mentors need to follow to properly test your additions. -->

- After pulling down the branch, navigate to your localhost in chrome and open the developers tools.

- Then navigate to the Application tab and expand the local storage file menu option and right click to clear your local user token (you can't get to the welcome screen without doing this step currently)
<img width="613" alt="Screen Shot 2020-06-20 at 2 37 28 AM" src="https://user-images.githubusercontent.com/2355543/85198682-7c1c6000-b29f-11ea-9a53-363d45fe3f0c.png">

- Then navigate to root in your browser and you should see this styled welcome screen.
![welcome-new](https://user-images.githubusercontent.com/2355543/85198699-9eae7900-b29f-11ea-8174-2df222ee2cd5.png)

meganmckissack committed:
bootstrap dependency uupdate

meganmckissack committed:
add material ui dependency

meganmckissack committed:
add material ui meta tag, baseline css to app.js, and import Roboto font

meganmckissack committed:
material ui experiments

Commit:
add image

Commit:
add image

Commit:
change css

Commit:
wallpaper experiments

Commit:
fix conflicts

meganmckissack committed:
added css styles, added background image, added shareList component to welcome page, removed footer nav from welcome page

meganmckissack committed:
created footer compoenent and tested routing on addItem.js, reverted footer styling but routing and styles are not working properly

meganmckissack committed:
uninstalled material ui core and icon dependencies

meganmckissack committed:
update flex css for main and remove bootstrap container

meganmckissack committed:
remove unused bootstrap container

meganmckissack commented:
I don't know why the checks are failing but there is an unused variable warning in the console even though it is used as a variable to add to the database.

`./src/Components/addItem.js
  Line 11:10:  'lastPurchasedDate' is assigned a value but never used  no-unused-vars`

meganmckissack commented:
~~Would love some tips for creating containers in flexbox.  The current code has the background resizing responsively is pushing the main content to fit rather than the main content resizing correctly.~~ 
Sort of fixed this issue after coming back to the problem with fresh eyes.

Commit:
add wallpaper pic

Commit:
fix container className

Commit:
add veg background pic

Commit:
merge conflicts

segdeha committed:
Fix builds

Builds no longer treat warnings as errors, per https://community.netlify.com/t/build-command-just-started-failing-create-react-app/17188/2?u=alejandronanez

meganmckissack committed:
fixed flex styling to wrap form element instead of main

DeeThompson commented:
I just ran the code and the front page looks great! Seriously I like it but then when I clicked on 'add an item' I got this, just wanted to know if we're seeing the same thing
![Screen Shot 2020-06-20 at 4 25 56 PM](https://user-images.githubusercontent.com/43711962/85211989-55503f00-b31c-11ea-9d20-bf37a2cd6aa6.png)

msholty-fd commented on src/Components/footerNav.js:
```diff
@@ -0,0 +1,21 @@
+
+function Footer() {
+  return (
+    <Router>
```
It may not be necessary to wrap this with `<Router />`, since you have context of the router via `<Router />` being used in `<App />` already.

msholty-fd commented on src/Components/shareList.js:
```diff
@@ -1,4 +1,6 @@
 import React, { useState } from 'react';
+import '../App.css';
```
I generally like to [organize my imports](https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/order.md) so as you maintain this file, it's easy to understand the types of dependencies it has on other parts of the codebase. Sometimes a file has a lot of third-party dependencies, or maybe local dependencies. Organizing them in groups like this will help you make sense of that easily!

msholty-fd reviewed (approved):
This looks great! I absolutely love the background image ❤️ I can tell a lot of thought was put into this, and I appreciate that. 🥇 

When I was on the grocery list page, where it lists out the items in my list, I couldn't find the footer anymore where I would be able to add an item. It seems like it's not rendering at all in the DOM. You should look into that before merging, so we don't lose that functionality. 

Otherwise, this looks awesome! 👍 ⚡

alejandronanez reviewed (approved):
I'm aligned with what @msholty-fd said, great work y'all - this is a night and day difference from the previous version - nice work!

I **loved** how you took care of improving accessibility and how you described your thought process!

I'm seeing some regressions though, apparently, the `remove item` functionality is gone and the list items have a transparent background, is that intentional? If not, I think we should get that fixed before merging!

Thanks 💯 !

![](https://cdn.zappy.app/a33a3d890aed0b052a0e3fee8b26bd0d.png)

meganmckissack committed:
add bg-gray to css for inactive list items

meganmckissack committed:
add footer element to grocery list page

meganmckissack committed:
fix footer navigation functionality and remove unecessary app.css import

meganmckissack committed:
tighten css for footer and background

meganmckissack committed:
remove footer from new token add item page

meganmckissack committed:
fix merge conflicts
